task8/ServletsOnMaven/src/main/webapp/resources/js/View.js

```diff
@@ -0,0 +1,1001 @@
+		new Post("23", "incredible4", new Date('2020-11-10T11:57:09'), "Hello46"),
+		new Post("24", "incredible4", new Date('2020-12-07T10:14:22'), "Hello111"),
+		new Post("25", "incredible4", new Date('2020-12-15T02:38:40'), "Hello481"),
+		new Post("26", "incredible4", new Date('2020-01-05T17:41:38'), "Hello211"),
+		new Post("27", "incredible4", new Date('2020-06-07T13:53:03'), "Hello154", ["#hello", "#hai"]),
+		new Post("28", "incredible4", new Date('2020-01-23T12:15:11'), "Hello564"),
+		new Post("29", "incredible4", new Date('2020-07-05T20:30:55'), "Hello931"),
+		new Post("30", "incredible4", new Date('2020-04-25T00:08:16'), "Hello990"),
+		new Post("31", "incredible5", new Date('2020-07-10T09:20:03'), "Hello633"),
+		new Post("32", "incredible5", new Date('2020-09-27T18:55:54'), "Hello468"),
+		new Post("33", "incredible5", new Date('2020-12-27T12:39:01'), "Hello530", ["#hai"]),
+		new Post("34", "incredible5", new Date('2020-12-20T18:54:23'), "Hello927", ["#ni hao"]),
+		new Post("35", "incredible5", new Date('2020-07-19T11:25:13'), "Hello549"),
+		new Post("36", "incredible5", new Date('2020-10-26T04:03:24'), "Hello616"),
+		new Post("37", "incredible5", new Date('2020-12-23T17:20:30'), "Hello682"),
+		new Post("38", "incredible5", new Date('2020-07-28T00:46:26'), "Hello208"),
+		new Post("39", "incredible5", new Date('2020-10-22T02:59:26'), "Hello940"),
+		new Post("40", "incredible5", new Date('2020-03-15T21:40:13'), "Hello999")
+	];
+	postsArray.forEach(item => {
+		idsOfPosts.push(item.id);
+		localStorage.setItem("Post with id=" + item.id, JSON.stringify(item));
+	});
+	localStorage.setItem("IDs of posts", JSON.stringify(idsOfPosts));
+	postsArray.forEach(x => {
+		if (!users.has(x.author)) {
+			users.set(x.author, new User(x.author, x.author));
+		}
+	});
+	localStorage.setItem("Users", JSON.stringify(Array.from(users)));
+	localStorage.setItem("Current user", JSON.stringify(null));
+	localStorage.setItem("Count of posts in feed", "10");
+	localStorage.setItem("IDs of current feed", JSON.stringify(idsOfPosts));
+})();*/
+
+function loadPostFromStorage(id) {
+	let post = JSON.parse(localStorage.getItem("Post with id=" + id));
+	return new Post(post.id, post.author, new Date(post.createdAt), post.description, post.hashTags, post.likes)
+}
+
+(function loadInfoFromStorage() {
+	idsOfPosts = JSON.parse(localStorage.getItem("IDs of posts"));
+	idsOfCurrentFeed = JSON.parse(localStorage.getItem("IDs of current feed"));
+	idsOfPosts.forEach(id => postsArray.push(loadPostFromStorage(id)));
+	JSON.parse(localStorage.getItem("Users")).forEach(item => users.set(item[0], new User(item[1].username, item[1].password)));
+})();
+
+let allPosts = new PostContainer(postsArray);
+
+class ViewPosts {
+
+	constructor() {
+		this.posts = new PostContainer();
+		idsOfCurrentFeed.forEach(id => {
+			this.posts.addPost(allPosts.getPost(id));
+		});
+		this.count = parseInt(JSON.parse(localStorage.getItem("Count of posts in feed")));
```

Comment:
Again, why it should be stored in LS? After reload, again 10 posts should be shown, but not the same amount as before reload